compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Utils/AMDAIELogicalObjFifoSplittingUtils.cpp

```diff
@@ -804,9 +804,7 @@ LogicalResult splitDoublyStridedOp(IRRewriter &rewriter,
                             ? maybeSplitFactor.value()
                             : std::gcd(sourceSize, targetSize);
   if (sourceSize % splitFactor != 0 || targetSize % splitFactor != 0) {
-    return op.emitOpError() << "the target or source size is not divisible by "
-                               "the provided splitting factor: "
-                            << splitFactor;
+    splitFactor = std::gcd(sourceSize, targetSize);
```

Comment:
This would be a good place to add a `LLVM_DEBUG` statement that mentions that the splitting factor is changed.